homeassistant/components/notion/diagnostics.py

```diff
@@ -0,0 +1,35 @@
+"""Diagnostics support for Notion."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+CONF_DEVICE_KEY = "device_key"
+
+TO_REDACT = {
+    CONF_DEVICE_KEY,
+    CONF_EMAIL,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    return {
+        "entry": {
+            "data": async_redact_data(entry.data, TO_REDACT),
+        },
```

Comment:
You could simple add the whole config entry (_check if further data needs to be redacted_)

```suggestion
        "entry": "async_redact_data(entry.as_dict(), REDACT_CONFIG),
```